lib/dspam/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+#################################################################################
+# CMakeLists.txt for DSPAM Library
+#################################################################################
+# This file builds the DSPAM spam filtering library which provides statistical
+# spam detection and filtering capabilities.
+
+# Library sources
+set(DSPAM_SOURCES
+    src/base64.c
+    src/bnr.c
+    src/buffer.c
+    src/config_shared.c
+    src/decode.c
+    src/diction.c
+    src/error.c
+    src/hash.c
+    src/hash_drv.c
+    src/heap.c
+    src/libdspam.c
+    src/list.c
+    src/nodetree.c
+    src/pref.c
+    src/read_config.c
+    src/tokenizer.c
+    src/util.c
+)
+
+# Create the DSPAM library
+add_library(mahogany_dspam ${DSPAM_SOURCES})
+
+# Set library properties
+set_target_properties(mahogany_dspam PROPERTIES
+    OUTPUT_NAME dspam
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+# Include directories
+target_include_directories(mahogany_dspam
+    PUBLIC
+        ${CMAKE_CURRENT_SOURCE_DIR}/src
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/src
+)
+
+# Preprocessor definitions
+target_compile_definitions(mahogany_dspam PRIVATE
+    _CRT_SECURE_NO_WARNINGS
+    _CRT_NONSTDC_NO_WARNINGS
+)
+
+# Create a basic config header if it doesn't exist
+if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/auto-config.h")
```

Comment:
The condition is useless, this file never exists and the file in the current binary directory must always be created.

Under Windows it must be created by copying `lib/dspam/src/auto-config.h.win32` to it.

Under Unix we need to use `lib/dspam/src/auto-config.h.in` as input file and generate output from it.